src/tests/Microsoft.PowerFx.Connectors.Tests/PowerPlatformConnectorTests.cs

```diff
@@ -85,6 +86,75 @@ public async Task MSNWeatherConnector_CurrentWeather(bool useSwaggerParameter)
             Assert.Equal(expected, actual);
         }
 
+        [Theory]
+        [InlineData(100)] // Continue
+        [InlineData(200)] // Ok
+        [InlineData(202)] // Accepted
+        [InlineData(305)] // Use Proxy
+        [InlineData(411)] // Length Required
+        [InlineData(500)] // Server Error
+        [InlineData(502)] // Bad Gateway
+        public async Task Connector_GenerateErrors(int statusCode)
+        {
+            using var testConnector = new LoggingTestServer(@"Swagger\TestConnector12.json");
+            var apiDoc = testConnector._apiDocument;
+
+            var config = new PowerFxConfig();
+
+            using var httpClient = new HttpClient(testConnector);
+            using var client = new PowerPlatformConnectorClient(
+                "firstrelease-001.azure-apim.net", // endpoint
+                "839eace6-59ab-4243-97ec-a5b8fcc104e4", // x-ms-client-environment-id
+                "8329fe1b70d8494e940a9d3f683e1845", // connectionId
+                () => "AuthToken",
+                httpClient)
+            {
+                SessionId = "4851caf7-23ec-43fc-9a56-e1628655a6bd" // from x-ms-request-url
+            };
+
+            var funcs = config.AddService("TestConnector12", apiDoc, client);
+
+            // Now execute it...
+            var engine = new RecalcEngine(config);
+            testConnector.SetResponse($"{statusCode}", (HttpStatusCode)statusCode);
+
+            var result = await engine.EvalAsync($"TestConnector12.GenerateError({{error: {statusCode}}})", CancellationToken.None);
+
+            Assert.NotNull(result);
+
+            if (statusCode < 300)
+            {
+                Assert.IsType<NumberValue>(result);
+
+                var nv = (NumberValue)result;
+
+                Assert.Equal(statusCode, nv.Value);
+            }
+            else
+            {
+                Assert.IsType<ErrorValue>(result);
+
+                var ev = (ErrorValue)result;
```

Comment:
>ErrorValue [](http://example.com/codeflow?start=26&length=10)

Errors still have a type. We should verify the type is NumberType.

Reply:
This matters in expressions like this, especially when the connector is returning a record:

IfError(MsnWeather.Get(...).field1,2)